api/v1beta2/ibmvpccluster_types.go

```diff
@@ -163,6 +178,45 @@ type IBMVPCClusterStatus struct {
 	Conditions capiv1beta1.Conditions `json:"conditions,omitempty"`
 }
 
+// VPCNetworkStatus provides details on the status of VPC network resources.
+type VPCNetworkStatus struct {
+	// resourceGroup references the Resource Group for Network resources for the cluster.
+	// This can be the same or unique from the cluster's Resource Group.
+	// +optional
+	ResourceGroup *ResourceStatus `json:"resourceGroup,omitempty"`
+
+	// vpc references the IBM Cloud VPC.
+	// +optional
+	VPC *VPCResourceStatus `json:"vpc,omitempty"`
+}
+
+// VPCResourceStatus identifies a VPC resource by id (and name) and whether it is ready.
+type VPCResourceStatus struct {
```

Comment:
`ResourceStatus` also look the same, can we reuse?

Reply:
I had designed the two status' initially, but can not recall why I had a unique one for VPC resources ATM.
I'll remove and use `ResourceStatus` for all resources instead.

Reply:
Removed to use `ResourceStatus`